types/signed_header.go

```diff
@@ -22,7 +26,41 @@ func (sH *SignedHeader) Verify(untrst header.Header) error {
 		// and is a programmer bug
 		panic(fmt.Errorf("%T is not of type %T", untrst, untrstH))
 	}
+	err := untrstH.ValidateBasic()
+	if err != nil {
+		return err
```

Comment:
You can wrap the error in the VerifyError struct here as well. Please take a look at the comment below.